src/IB/IIMethod.cpp

```diff
@@ -233,6 +233,58 @@ IIMethod::getFEDataManager(const unsigned int part) const
     return d_fe_data_managers[part];
 } // getFEDataManager
 
+NumericVector<double>*
+IIMethod::getMeshCoordinates(bool is_current_configuration, std::string time, unsigned int part)
+{
+    if (is_current_configuration)
+    {
+        NumericVector<double>* X_ghost_vec = d_X_IB_ghost_vecs[part];
+        NumericVector<double>* X_vec = nullptr;
+
+        if (time == "current")
+        {
+            X_vec = d_X_current_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec);
+        }
+        else if (time == "half")
+        {
+            X_vec = d_X_half_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec);
+        }
+        else if (time == "new")
+        {
+            X_vec = d_X_new_vecs[part];
+            copy_and_synch(*X_vec, *X_ghost_vec);
+        }
+        else if (time == "ib_ghost")
+        {
+            // already assigned X_ghost_vec
+        }
+        else
+        {
+            TBOX_ERROR("Time must be set to current, half, new, or ib_ghost. Case sensitive.");
+        }
+        return X_ghost_vec;
+    }
+    else
+    { // reference configuration
+        EquationSystems* equation_systems = d_fe_data_managers[part]->getEquationSystems();
+        const MeshBase& mesh = equation_systems->get_mesh();
+        const unsigned int dim = mesh.mesh_dimension();
+        std::unique_ptr<QBase> qrule;
+        System& X_system = equation_systems->get_system(COORDS_SYSTEM_NAME);
+        const DofMap& X_dof_map = X_system.get_dof_map();
+        FEDataManager::SystemDofMapCache& X_dof_map_cache =
+            *d_fe_data_managers[part]->getDofMapCache(COORDS_SYSTEM_NAME);
+        FEType X_fe_type = X_dof_map.variable_type(0);
+        NumericVector<double>& X0_vec_ref = X_system.get_vector("INITIAL_COORDINATES");
+        NumericVector<double>* X0_vec = &X0_vec_ref;
+        return X0_vec;
```

Comment:
1. You just need the vector of DOFs and not all of the other stuff here.
2. FWIW, initial coordinates are not necessarily the reference coordinates --- there might be an initial transformation applied from the mesh coordinates (which we can take to be reference coordinates) and the initial coordinates.